src/sentry/integrations/slack/notify_action.py

```diff
@@ -162,31 +162,25 @@ def get_channel_id(self, integration_id, name):
 
         # Look for channel ID
         channels_payload = dict(
-            token_payload, **{"exclude_archived": False, "exclude_members": True}
+            token_payload, **{"exclude_archived": False, "types": "public_channel,private_channel", "limit": 1000}
         )
 
-        resp = session.get("https://slack.com/api/channels.list", params=channels_payload)
-        resp = resp.json()
-        if not resp.get("ok"):
-            self.logger.info("rule.slack.channel_list_failed", extra={"error": resp.get("error")})
-            return None
-
-        channel_id = {c["name"]: c["id"] for c in resp["channels"]}.get(name)
-
-        if channel_id:
-            return (CHANNEL_PREFIX, channel_id)
+        cursor = None
+        while True:
+            resp = session.get("https://slack.com/api/conversations.list", params=dict(channels_payload, **{"cursor": cursor}))
+            resp = resp.json()
+            if not resp.get("ok"):
+                self.logger.info("rule.slack.conversation_list_failed", extra={"error": resp.get("error")})
+                return None
 
-        # Channel may be private
-        resp = session.get("https://slack.com/api/groups.list", params=channels_payload)
-        resp = resp.json()
-        if not resp.get("ok"):
-            self.logger.info("rule.slack.group_list_failed", extra={"error": resp.get("error")})
-            return None
+            channel_id = {c["name"]: c["id"] for c in resp["channels"]}.get(name)
 
-        group_id = {c["name"]: c["id"] for c in resp["groups"]}.get(name)
+            if channel_id:
+                return (CHANNEL_PREFIX, channel_id)
 
-        if group_id:
-            return (CHANNEL_PREFIX, group_id)
+            cursor = resp.get("response_metadata", {}).get("next_cursor", "")
+            if not cursor:
```

Comment:
nit: Could you change this to `.get("next_cursor")` and `if cursor is None:`? Just since we don't actually use the empty string for anything other than the condtional